src/printer.js

```diff
@@ -1271,32 +1271,36 @@ function handleNewSetLiteral(path, print) {
 }
 
 function handleNewListInit(path, print) {
-  // Technically there are 2 ways to declare new list:
-  // - new String[]{};
-  // - new List<String>{};
-  // We use the 2nd format because it can handle any type of new list, e.g.
-  // This is correct:
-  // List<List<String>> data = new List<List<String>>{};
-  // But this is not:
-  // List<List<String>> data = new List<String>[]{};
-
-  // Other quirks:
-  // All of these are correct:
-  // Id[] ids = new Id[]{};
-  // Id[] ids = new Id[1];
-  // Id[] ids = new Id[anotherIds];
-  // But this is not correct:
-  // Id[] ids = new Id[1]{};
+  // We can delcare lists in the following ways:
+  // new Object[size];
+  // new Object[] { value, ... };
+  // new List<Object>(); // Provides AST consistency.
+  // new List<Object>(size);
+
+  // We use Object[size] if a literal number is provided.
+  // We use List<Object>(param) otherwise.
+  // This should provide compatibility for all known types without knowing
+  // if the parameter is a variable (copy constructor) or literal size.
+
   const expressionDoc = path.call(print, "expr", "value");
   const parts = [];
+  const typePart = path.map(print, "types");
+  const hasLiteralNumberInitializer =
+    (typePart.group || (typePart.length && typePart[0].parts.length < 4)) &&
+    !Number.isNaN(parseInt(expressionDoc, 10));
```

Comment:
The AST does give us information about the literal type, so we don't need to parse this on our own. For example, this code will return whether the param type is Integer:

```
const node = path.getValue();
return node.expr && node.expr.type && node.expr.type['$'] === 'INTEGER';
```

Reply:
Thanks, I wasn't able to figure this one out with the apparently sparse documentation. I'm rather pro at browser debugging, not so much much CLI-based debugging.